### PR TITLE
fix(tauri) event system init on windows fixed

### PR DESCRIPTION
**Pull request by lucasfernog**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] New Binding Issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [ ] No

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch and _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/tauri/tree/dev/docs) (for faster update click on "Suggest an edit on GitHub" at bottom of page) or explained in the PR's description.

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: lib/rust/src/api.rs [hunk 1]
```diff
@@ -9,6 +9,46 @@ pub fn handler<T: 'static>(webview: &mut WebView<'_, T>, arg: &str) -> bool {
...
+                      const listeners = (window['{listeners}'] && window['{listeners}'][payload.type]) || []
+
+                      if (!ignoreQueue && listeners.length === 0) {{ 
+                        window['{queue}'].push({{ 
+                          payload: payload,
+                          salt: salt
+                        }})
+                      }}
+
+                      for (let i = listeners.length - 1; i >= 0; i--) {{
```
Comment (nothingismagick) [inline lib/rust/src/api.rs]: var not let ?

File: lib/rust/src/api.rs [hunk 2]
```diff
@@ -9,6 +9,46 @@ pub fn handler<T: 'static>(webview: &mut WebView<'_, T>, arg: &str) -> bool {
...
+            .dispatch(move |_webview| {
+              _webview
+                .eval(&format!(
+                  "window['{queue}'] = [];
+                  window['{fn}'] = function (payload, salt, ignoreQueue) {{
+                    window.tauri.promisified({{
+                      cmd: 'validateSalt',
+                      salt: salt
+                    }}).then(function () {{
+                      const listeners = (window['{listeners}'] && window['{listeners}'][payload.type]) || []
```
Comment (nothingismagick) [inline lib/rust/src/api.rs]: var not const?

File: lib/rust/src/api.rs [hunk 3]
```diff
@@ -9,6 +9,46 @@ pub fn handler<T: 'static>(webview: &mut WebView<'_, T>, arg: &str) -> bool {
...
+            .handle()
+            .dispatch(move |_webview| {
+              _webview
+                .eval(&format!(
+                  "window['{queue}'] = [];
+                  window['{fn}'] = function (payload, salt, ignoreQueue) {{
+                    window.tauri.promisified({{
+                      cmd: 'validateSalt',
+                      salt: salt
+                    }}).then(function () {{
```
Comment (nothingismagick) [inline lib/rust/src/api.rs]: .then -> maybe we should turn this into callback style?

File: lib/tauri.js [hunk 4]
```diff
@@ -7,7 +7,7 @@
  **/
 
 // open <a href="..."> links with the Tauri API
-document.querySelector('body').addEventListener('click', e => {
+document.querySelector('body').addEventListener('click', function (e) {
```
Comment (nothingismagick) [inline lib/tauri.js]: 👍

File: lib/rust/src/api.rs (hunk 3 above)
Comment (nothingismagick) [inline lib/rust/src/api.rs]: then is es6, no?
this means we should probably rewrite this whole thing with es5 style callbacks, no?

File: lib/rust/src/api.rs (hunk 1 above)
Comment (nothingismagick) [inline lib/rust/src/api.rs]: s/let/var

File: lib/rust/src/api.rs [hunk 5]
```diff
@@ -9,6 +9,46 @@ pub fn handler<T: 'static>(webview: &mut WebView<'_, T>, arg: &str) -> bool {
...
+
+                      if (!ignoreQueue && listeners.length === 0) {{ 
+                        window['{queue}'].push({{ 
+                          payload: payload,
+                          salt: salt
+                        }})
+                      }}
+
+                      for (let i = listeners.length - 1; i >= 0; i--) {{ 
+                        const listener = listeners[i]
```
Comment (nothingismagick) [inline lib/rust/src/api.rs]: s/const/var

File: lib/rust/src/api.rs (hunk 3 above)
Comment (lucasfernog) [reply lib/rust/src/api.rs]: promise support is available from quasar polyfill

**Pull request merged**